### PR TITLE
[Bugfix] Fixes 'SLEEPING' vehicles spawn at (0,0,0)

### PR DESCRIPTION
**Pull request by ulteq**
Fixes: #1047

### CODE REVIEW

File: source/main/physics/BeamFactory.cpp [hunk 1]
```diff
@@ -974,11 +974,15 @@ void BeamFactory::update(float dt)
...
-				m_trucks[t]->sendStreamData();
-			if (m_trucks[t]->state < SLEEPING) {
+			if (m_trucks[t]->state < SLEEPING)
 			    m_trucks[t]->UpdatePropAnimations(dt);
-                        }
+			if (m_trucks[t]->networking)
+			{
+				if (m_trucks[t]->state == SIMULATED)
+					m_trucks[t]->sendStreamData();
+				if (m_trucks[t]->state == SLEEPING && m_trucks[t]->netTimer.getMilliseconds() < 10000)
```
Comment (only-a-ptr) [inline source/main/physics/BeamFactory.cpp]: Where did the 10s timeout come from? :grimacing:

**Comment by ulteq** [comment]
> This doesn't change anything which affects vehicle position - what's going on here?

This only affects `NETWORKED` trucks in multiplayer.

> Where did the 10s timeout come from?

Gut instincts

**Comment by only-a-ptr** [comment]
I don't understand this fix. It appears to stop position updates for 10sec each time the vehicle goes to sleep. But how does it help with spawn position?

**Comment by ulteq** [comment]
This code only runs on the client that owns the truck.

And it causes the client to send stream data for `SLEEPING` trucks during the first 10 seconds of their lifetime.

**Comment by only-a-ptr** [comment]
:+1:

**Pull request merged**